File: Sensor.java
import java.util.ArrayList;

public abstract class Sensor {
    public Sensor(int z){
        this(z, SwitchState.CLOSE);
    }
    public Sensor(int z, SwitchState s){
        zone = z;
        state = s;
    }

    public SwitchState getState(){
        return state;
    }
    public int getZone() {
        return zone;
    }
    protected void setState(SwitchState s) {
        state = s;
    }

    public void setZone(int zone){
        this.zone = zone;
    }
    private SwitchState state;
    private int zone;
}

File: sensor.cpp
#include "sensor.h"
#include <iostream>
//using namespace std;

Sensor::Sensor(int z, bool close){
    this->zone = z;
    this->close = close;
}
bool Sensor::isClose() const {
    return close;
}
int Sensor::getZone() const {
    return zone;
}
void Sensor::setClose(bool isClose) {
    this->close = isClose;
    std::cout << "Sensor is "<< (close?"close.":"open.") << std::endl;
}
